src/Microsoft.Sbom.Parsers.Spdx30SbomParser/Generator.cs

```diff
@@ -524,18 +524,22 @@ private List<Element> GetSpdxRelationshipsFromSbomFile(Element spdxFileElement,
         spdxRelationshipLicenseConcludedElement.AddSpdxId();
         spdxRelationshipAndLicenseElementsToAddToSBOM.Add(spdxRelationshipLicenseConcludedElement);
 
-        // If they exist, convert licenseDeclared to SPDX license elements and add Relationship elements for them
-        if (fileInfo.LicenseInfoInFiles == null || !fileInfo.LicenseInfoInFiles.Any())
-        {
-            return spdxRelationshipAndLicenseElementsToAddToSBOM;
-        }
-
         var toRelationships = new List<string>();
-        foreach (var licenseInfoInOneFile in fileInfo.LicenseInfoInFiles)
+        if (fileInfo.LicenseInfoInFiles == null)
         {
-            var licenseDeclaredElement = GenerateLicenseElement(licenseInfoInOneFile);
+            var licenseDeclaredElement = GenerateLicenseElement(null);
```

Comment:
This always generates a new `NoAssertionElement`. Would it be useful to create a single immutable `NoAssertionElement` that we just use over and over? The "immutable" part may make this more trouble that it's worth

Reply:
That looks OK, as long as we're confident that we never change its properties. The underlying `Element` object is not immutable, so it creates a potential for an unintended change to apply to multiple places